### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by shafran-andriy**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,78 @@
...
+    def __init__(self: "Distance", km: int) -> None:
+        self.km = km
+
+    def __str__(self: "Distance") -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self: "Distance") -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self: "Distance", other: "Distance") -> "Distance":
```
Comment (EdAlekseiev) [inline app/main.py:11]: other can be int and float. Fix everywhere
```suggestion
    def __add__(self: "Distance", other: "Distance" | int | float) -> "Distance":
```

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (shafran-andriy) [reply app/main.py]: Fixed

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,85 @@
...
+    def __init__(self: "Distance", km: int) -> None:
+        self.km = km
+
+    def __str__(self: "Distance") -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self: "Distance") -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self: "Distance",
```
Comment (vasylhnatiuk) [inline app/main.py:14]:
```suggestion
    def __add__(self: Distance,
```

Comment (vasylhnatiuk) [reply app/main.py:14]: use from __future__ import annotations

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 2 above)
Comment (shafran-andriy) [reply app/main.py]: fixed

**Review by YukihiroSM — CHANGES_REQUESTED** [review]

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**